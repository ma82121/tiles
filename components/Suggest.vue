<template>
  <div v-if="isShow" class="suggest">
    <div class="suggest-message">
      {{ message }}
    </div>
  </div>
</template>

<script>
export default {
  name: 'Suggest',
  props: {
    isShow: {
      type: Boolean,
      default: true
    },
    message: {
      type: String,
      default: 'ヒントあげようかな。どうしようかなー'
    }
  }
}
</script>

<style lang="scss" scoped>
.suggest {
  width: 100%;
  height: 60px;
  display: flex;
  align-items: center;
  justify-content: center;
  word-wrap: break-word;
  overflow-wrap: break-word;
}

.suggest-message {
  position: relative;
  padding: 10px;
  width: 300px;
  height: 60px;
  box-sizing: border-box;
  border-radius: 4px;
  background-color: #fff;
  font-size: 0.8rem;
  color: #1cc4fd;
  border: 1px solid #1cc4fd;
}
</style>
