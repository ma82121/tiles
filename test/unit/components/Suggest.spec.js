import { shallowMount, createLocalVue } from '@vue/test-utils'
import Vuex from 'vuex'
import Suggest from '@/components/Suggest'

const localVue = createLocalVue()
localVue.use(Vuex)

let store
beforeEach(() => {
  store = new Vuex.Store({
    state: {
      config: {
        useSuggest: true
      }
    }
  })
})

describe('Suggest', () => {
  test('is a vue instance', () => {
    const message = '10手詰'
    const wrapper = shallowMount(Suggest, {
      store,
      localVue,
      propsData: {
        message
      }
    })
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  test('renders props.message when passed', () => {
    const message = '10手詰'
    const wrapper = shallowMount(Suggest, {
      store,
      localVue,
      propsData: {
        message
      }
    })
    const div = wrapper.find('.suggest-message')
    expect(div.text()).toBe(message)
  })

  test('snapshot', () => {
    const message = '10手詰'
    const wrapper = shallowMount(Suggest, {
      store,
      localVue,
      propsData: {
        message
      }
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})
