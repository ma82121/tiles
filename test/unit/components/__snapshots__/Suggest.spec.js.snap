// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Suggest snapshot 1`] = `
<div
  class="suggest"
>
  <div
    class="suggest-message"
  >
    
    10手詰
  
  </div>
</div>
`;
